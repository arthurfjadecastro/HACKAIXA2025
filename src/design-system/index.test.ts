import * as designSystemIndex from './index';

describe('Design System Index', () => {
  it('should export design system modules', () => {
    expect(designSystemIndex).toBeDefined();
    expect(typeof designSystemIndex).toBe('object');
  });

  it('should re-export Icon components', () => {
    expect(designSystemIndex.Icon).toBeDefined();
    expect(designSystemIndex.useIcon).toBeDefined();
    expect(designSystemIndex.iconNames).toBeDefined();
    expect(designSystemIndex.iconSizes).toBeDefined();
  });

  it('should re-export tokens and components', () => {
    // Verifica se mÃ³dulos foram exportados sem erros
    expect(() => {
      const exported = { ...designSystemIndex };
      expect(exported).toBeDefined();
    }).not.toThrow();
  });

  it('should match snapshot', () => {
    expect(Object.keys(designSystemIndex).sort()).toMatchSnapshot();
  });
});
