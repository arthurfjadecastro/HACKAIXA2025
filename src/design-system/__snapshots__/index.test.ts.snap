// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System Index should match snapshot 1`] = `
[
  "Accordion",
  "ActionCard",
  "ArthurAvatar",
  "BottomSheet",
  "BottomSheetHeader",
  "Button",
  "Card",
  "Divider",
  "EmptyState",
  "HeroHeader",
  "Icon",
  "InputField",
  "Skeleton",
  "Text",
  "borderRadius",
  "colors",
  "elevation",
  "fontSizes",
  "fontWeights",
  "gap",
  "iconNames",
  "iconSizes",
  "lineHeights",
  "margin",
  "padding",
  "radius",
  "shadows",
  "spacing",
  "theme",
  "typography",
  "useIcon",
]
`;
